### PR TITLE
CurtisWoodworth/Alarm

### PR DESCRIPTION
**Pull request by CurtisWoodworth**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ repos:
         additional_dependencies: [toml]
         # need to exclude files here as isort in pre-commit hook doesn't seem
         # to look for config in pyproject.toml
-        exclude: (rokku\.py|raspberry_pi_camera\/tests\/|raspberry_pi_motion_sensor\/tests|raspberry_pi_ui\/utility\.py)
+        exclude: (rokku\.py|raspberry_pi_camera\/tests\/|raspberry_pi_motion_sensor\/tests\/|raspberry_pi_alarm\/tests\/|raspberry_pi_ui\/utility\.py)
```
Comment (FanchenBao) [inline .pre-commit-config.yaml]: Nice! Good observation!

File: src/raspberry_pi_ui/buttons/alarm.py [hunk 2]
```diff
@@ -31,11 +31,75 @@ def __init__(self, button, pub, msg_q):
...
+                self.alarm_sounding = False
+
+            # If message from rpi_out was recieved
+            if self.alarm_sounding_out:
+                # buzzer is playing on rpi_out: turn button to red
+                self.logger.info("Alarm SOUNDING on rpi_out")
+                set_button_property(self, "red", "Silence Alarm")
+            else:  # A message from rpi_out was not recieved
+                self.logger.error(
+                    f"Motion status: rpi_in = {self.alarm_sounding}, rpi_out = {self.alarm_sounding_out}"
```
Comment (FanchenBao) [inline src/raspberry_pi_ui/buttons/alarm.py]: Probably should be "Alarm status"

File: src/raspberry_pi_ui/buttons/alarm.py [hunk 3]
```diff
@@ -31,11 +31,75 @@ def __init__(self, button, pub, msg_q):
...
+                self.alarm_sounding = True
+
+            if not self.alarm_sounding:
+                # rpi_out message recieved, motion detection is off, turn button to blue
+                self.logger.info("Alarm is SILENCED on rpi_out")
+                set_button_property(self, "blue", "Sound Alarm")
+                self.alarm_sounding = False
+            else:  # A message from rpi_out was not recieved
+                self.logger.error(
+                    f"Motion status: rpi_in = {self.alarm_sounding}, rpi_out = {self.alarm_sounding_out}"
```
Comment (FanchenBao) [inline src/raspberry_pi_ui/buttons/alarm.py]: Same as above, probably should be "Alarm status"

File: src/raspberry_pi_ui/buttons/alarm.py (hunk 2 above)
Comment (CurtisWoodworth) [reply src/raspberry_pi_ui/buttons/alarm.py]: Heh, yeah good catch.

**Pull request merged**